src/rapidhtml/tags.py

```diff
@@ -126,6 +142,99 @@ def render(self):
 
         return ret_html
 
+    def select(
+        self,
+        select_tag: typing.Type["BaseTag"] | str,
+        *,
+        recurse: bool = False,
+        pop: bool = False,
+        first_match: bool = False,
+    ) -> list["BaseTag"]:
+        """
+        Selects and returns a list of BaseTag objects that match the given tag name or BaseTag object.
+
+        Args:
+            tag (Type[BaseTag] or str): The uninstantiated subclass of BaseTag or the tag name to match against.
+            recurse (bool, optional): If True, recursively searches for matching tags within nested BaseTag objects. Defaults to False.
+            pop (bool, optional): If True, removes the selected tags from the current object. Defaults to False.
+            first_match (bool, optional): If True, stops searching after finding the first match. Defaults to False.
+
+        Returns:
+            list[BaseTag]: A list of BaseTag objects that match the given tag name or BaseTag object.
+
+        Raises:
+            KeyError: If no matching tags are found and pop is True.
+            ValueError: If an instantiated object is passed as the tag parameter.
+            TypeError: If the tag parameter is of an unexpected type.
+
+        """
+        if isinstance(select_tag, BaseTag):
+            raise ValueError("Cannot pass instantiated object to select method!")
+        elif inspect.isclass(select_tag) and issubclass(select_tag, BaseTag):
+            select_tag_name = select_tag.__name__.lower()
+        elif isinstance(select_tag, str):
+            select_tag_name = select_tag
+        else:
+            raise TypeError(f"Unexpected tag type {type(select_tag)}. {select_tag}")
+
+        ret_tags: list["BaseTag"] = []
+        self_matches: list[tuple[int, "BaseTag"]] = []
+
+        for i, tag in enumerate(self.tags):
+            if not isinstance(tag, BaseTag):
+                continue
+            tag_name = tag.tag_name
+            if tag_name == select_tag_name or tag_name == f"htmltag{select_tag_name}":
```

Comment:
As part of my callback PR, I actually normalized these tag names, so the latter part of this check isn't necessary
https://github.com/ONTechNITS/RapidHTML/blob/ec8ece5f9e1fb8032d7a4a90f17d024c646cc1bf/src/rapidhtml/tags.py#L50